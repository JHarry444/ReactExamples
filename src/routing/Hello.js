import React from 'react';

export default function Hello(props) {
    return (
        <p>Hello, the id is {props.match}</p>
    );
}