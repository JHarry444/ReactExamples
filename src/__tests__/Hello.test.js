import React from 'react';
import Hello from '../routing/Hello';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
    const tree = shallow(<Hello />);
    tree.update()
    expect(tree).toMatchSnapshot();
});


it('contains correct id', () => {
    const bloop = 1;
    const hello = shallow(<Hello match={bloop} />);
    expect(hello.containsMatchingElement(<p>Hello, the id is 1</p>)).toBeTruthy();
    expect(hello).toMatchSnapshot();
})

