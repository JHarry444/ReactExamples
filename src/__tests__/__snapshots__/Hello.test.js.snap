// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contains correct id 1`] = `
<p>
  Hello, the id is 
  1
</p>
`;

exports[`renders without crashing 1`] = `
<p>
  Hello, the id is 
</p>
`;
